### PR TITLE
SVG i18n messes up DOM

### PR DESCRIPTION
**Pull request by raphink**
Here's a simple example:

```svg
<?xml version="1.0" encoding="UTF-8" standalone="no"?>
<svg xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink" width="60" height="20">
  <g transform="translate(10,20)">
    <text><tspan style="fill:#ff0000">{{test}}</tspan></text>
  </g>
</svg>
```

Add a `test` entry in `locales/string.json` with `{ "test": "My test"}`. The text gets replaced, but the style is lost because the `<tspan>` element is removed from the SVG.

### CODE REVIEW

**Comment by raphink** [comment]
Just attached some code which fixes this by:

* translating `tspan` nodes (but we might want to keep extending this list)
* only selecting immediate text nodes, not nested nodes (which avoids replacing intermediary nodes)

File: public/js/simpleStrings-0.0.1.js [hunk 1]
```diff
@@ -31,7 +31,7 @@
 
       function translate(item) {
         item = $(item);
-        var text = item.text();
+        var text = item.clone().children().remove().end().text();
```
Comment (binford2k) [inline public/js/simpleStrings-0.0.1.js:34]: Ah! I think I've finally figured out why you're doing this. Given this example,

```svg
<?xml version="1.0" encoding="UTF-8" standalone="no"?>
<svg xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink" width="60" height="20">
  <g transform="translate(10,20)">
    <text><tspan style="fill:#ff0000">{{red}}</tspan><tspan style="fill:#00ff00">{{green}}</tspan></text>
  </g>
</svg>
```

The original code would have matched the outermost `<text>` tag and evaluated the text value it and all children, or `{{red}}{{green}}`. This code means that that match would have evaluated *nothing*, but then the `<tspan>` tags would then each be individually matched and then evaluated.

Is that a correct assessment of the intent? If so, I'll totally merge this. This is a better solution than my original which required you to make separate text elements.

**Review by binford2k — APPROVED** [review]

File: public/js/simpleStrings-0.0.1.js (hunk 1 above)
Comment (binford2k) [reply public/js/simpleStrings-0.0.1.js:34]: Actually, no. Am going to have to think about this more. This solution will work perfectly well when there’s no actual text in the text element, but as soon as there’s a replacement token that’s outside the `tspan` element(s), then the call to `text(replacement)` will replace all children🙃.

Lemme think on this a bit more. If we required that the token filled the entire element, we’d probably be ok, but I can see why you want to enable multiple tokens in a string.

Comment (binford2k) [reply public/js/simpleStrings-0.0.1.js:34]: I'm going to rewrite this whole routine using `.contents()` to get the text nodes as well as any children. Then I'll just translate them individually. Thanks for running down this rabbit hole with me., I think we'll come out of this with a solid solution.

Comment (raphink) [reply public/js/simpleStrings-0.0.1.js:34]: ok, I'm fine with that :wink:

**Comment by binford2k** [comment]
Fixes incorporated into #852

**Pull request closed without merging**